### PR TITLE
Add mujoco reacher

### PR DESCRIPTION
**Pull request by LucasAlegre**
Issue #22

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,6 +62,7 @@ You can also check more examples in this colab notebook!
...
 | `mo-lunar-lander-v2` <br><img src="https://raw.githubusercontent.com/Farama-Foundation/MO-Gymnasium/main/screenshots/lunarlander.png" width="200px">                 | Continuous / Discrete or Continuous | `[landed, shaped_reward, main_engine_fuel, side_engine_fuel]` | MO version of the "LunarLander-v2" environment. Objectives defined similarly as in [Hung et al. 2022](https://openreview.net/forum?id=AwWaBXLIJE).                                                                                                                                                         |
 | `mo-reacher-v0` <br><img src="https://raw.githubusercontent.com/Farama-Foundation/MO-Gymnasium/main/screenshots/reacher.png" width="200px">                          | Continuous / Discrete               | `[target_1, target_2, target_3, target_4]`                    | Reacher robot from [PyBullet](https://github.com/benelot/pybullet-gym/blob/ec9e87459dd76d92fe3e59ee4417e5a665504f62/pybulletgym/envs/roboschool/robots/manipulators/reacher.py), but there are 4 different target positions. From [Alegre et al. 2022](https://proceedings.mlr.press/v162/alegre22a.html). |
```
Comment (ffelten) [inline README.md:64]: Should we add a deprecated note or somth?

File: mo_gymnasium/envs/mujoco/__init__.py [hunk 2]
```diff
@@ -12,3 +12,9 @@
     entry_point="mo_gymnasium.envs.mujoco.hopper:MOHopperEnv",
     max_episode_steps=1000,
 )
+
+register(
+    id="mo-reacher-v4",
+    entry_point="mo_gymnasium.envs.mujoco.reacher:MOReacherEnv",
+    max_episode_steps=50,
```
Comment (ffelten) [inline mo_gymnasium/envs/mujoco/__init__.py:19]: I think the pybullet version has 100 steps max

File: mo_gymnasium/envs/mujoco/reacher.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+        self._step_mujoco_simulation(real_action, self.frame_skip)
+        if self.render_mode == "human":
+            self.render()
+
+        ob = self._get_obs()
+        return (
+            ob,
+            vec_reward,
+            False,
```
Comment (ffelten) [inline mo_gymnasium/envs/mujoco/reacher.py:55]: It never finishes?

File: mo_gymnasium/envs/mujoco/reacher.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+from gymnasium import utils
+from gymnasium.envs.mujoco import MujocoEnv
+from gymnasium.envs.mujoco.reacher_v4 import ReacherEnv
+from gymnasium.spaces import Box, Discrete
+
+
+DEFAULT_CAMERA_CONFIG = {"trackbodyid": 0}
+
+
+class MOReacherEnv(ReacherEnv):
```
Comment (ffelten) [inline mo_gymnasium/envs/mujoco/reacher.py:13]: Could you add pydoc with some high level explanation of what the env is about?

**Comment by ffelten** [comment]
- [ ] add tests

File: mo_gymnasium/envs/mujoco/__init__.py (hunk 2 above)
Comment (LucasAlegre) [reply mo_gymnasium/envs/mujoco/__init__.py:19]: I think it is better to use 'Reacher-V4' limit. This env is not going to be an exact reproduction of the PyBullet version anyway.

File: mo_gymnasium/envs/mujoco/reacher.py (hunk 3 above)
Comment (LucasAlegre) [reply mo_gymnasium/envs/mujoco/reacher.py:55]: The TimeLimit wrapper added by `mo_gym.make` that handles the `truncated` flag.

File: README.md (hunk 1 above)
Comment (LucasAlegre) [reply README.md]: What do you think now?

File: mo_gymnasium/envs/mujoco/reacher.py (hunk 4 above)
Comment (LucasAlegre) [reply mo_gymnasium/envs/mujoco/reacher.py:13]: I'll address this in a new PR to solve #24

**Pull request merged**